backend/dataall/modules/datasets/db/dataset_profiling_repository.py

```diff
@@ -0,0 +1,126 @@
+from sqlalchemy import and_
+
+from dataall.db import paginate, models
+from dataall.db.exceptions import ObjectNotFound
+from dataall.modules.datasets_base.db.models import DatasetProfilingRun, DatasetTable, Dataset
+
+
+class DatasetProfilingRepository:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def save_profiling(session, dataset, env, glue_table_name):
+        run = DatasetProfilingRun(
+            datasetUri=dataset.datasetUri,
+            status='RUNNING',
+            AwsAccountId=env.AwsAccountId,
+            GlueJobName=dataset.GlueProfilingJobName or 'Unknown',
+            GlueTriggerSchedule=dataset.GlueProfilingTriggerSchedule,
+            GlueTriggerName=dataset.GlueProfilingTriggerName,
+            GlueTableName=glue_table_name,
+            GlueJobRunId=None,
+            owner=dataset.owner,
+            label=dataset.GlueProfilingJobName or 'Unknown',
+        )
+
+        session.add(run)
+        session.commit()
+        return run
+
+    @staticmethod
+    def update_run(session, run_uri, glue_job_run_id):
+        run = DatasetProfilingRepository.get_profiling_run(
+            session, profiling_run_uri=run_uri, glue_job_run_id=glue_job_run_id
+        )
+        if glue_job_run_id:
+            run.GlueJobRunId = glue_job_run_id
+        session.commit()
+        return run
+
+    @staticmethod
+    def get_profiling_run(
+        session, profiling_run_uri=None, glue_job_run_id=None, glue_table_name=None
+    ):
+        if profiling_run_uri:
+            run: DatasetProfilingRun = session.query(
+                DatasetProfilingRun
+            ).get(profiling_run_uri)
+        else:
+            run: DatasetProfilingRun = (
+                session.query(DatasetProfilingRun)
+                .filter(DatasetProfilingRun.GlueJobRunId == glue_job_run_id)
+                .filter(DatasetProfilingRun.GlueTableName == glue_table_name)
+                .first()
+            )
+        return run
+
+    @staticmethod
+    def list_profiling_runs(session, dataset_uri):
+        # TODO filter is always default
+        filter = {}
+        q = (
+            session.query(DatasetProfilingRun)
+            .filter(DatasetProfilingRun.datasetUri == dataset_uri)
+            .order_by(DatasetProfilingRun.created.desc())
+        )
+        return paginate(
+            q, page=filter.get('page', 1), page_size=filter.get('pageSize', 20)
+        ).to_dict()
+
+    @staticmethod
+    def list_table_profiling_runs(session, table_uri):
+        # TODO filter is always default
```

Comment:
I have written this TODO to drag some awareness that there might be a potential bug: we don't show more than 20 results. Those results are latest, so it might be expected, but it's very confusing - if we don't need the filter at all I'd rather use limit and not a pagination (and renamed some methods like `list_latest_table_profiling_runs`